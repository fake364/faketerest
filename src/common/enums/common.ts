export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
