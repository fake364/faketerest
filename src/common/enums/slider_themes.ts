export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
