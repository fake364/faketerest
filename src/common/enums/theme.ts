export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}