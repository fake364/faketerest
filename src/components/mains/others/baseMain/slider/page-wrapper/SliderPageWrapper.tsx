import { BouncingButtons } from '../../bouncing-buttons/BouncingButtons';
import clsx from 'clsx';
import TitleSwitcher from '../title-switcher/TitleSwitcher';
import PointsContainer from '../points-container/PointsContainer';
import ImageFlexContainer from '../../image-grid/ImageFlexContainer';
import React, { FC } from 'react';
import { SLIDER_THEMES } from '../../../../../../common/enums/slider_themes';
import SliderNode, {
  SLIDER_ARRAY
} from '../../../../../../common/classes/customClasses/SliderNode';

const sliderNodes = SliderNode.createThemesArray(SLIDER_ARRAY);

type Props = {
  animated: boolean;
  shownType: SLIDER_THEMES;
  pageSlidingStopped: boolean;
  onClickDown: () => void;
  onClickUp: () => void;
};

const SliderPageWrapper: FC<Props> = ({
  pageSlidingStopped,
  animated,
  onClickUp,
  onClickDown,
  shownType
}) => (
  <>
    <BouncingButtons
      isAnimated={animated}
      shownType={shownType}
      onClickDown={onClickDown}
      onClickUp={onClickUp}
    />
    <div
      className={clsx(
        'w-full flex flex-col items-stretch relative z-[-1]',
        'animate-slide_bottom',
        pageSlidingStopped && 'stop-animation'
      )}
    >
      <TitleSwitcher shownType={shownType} isAnimation={animated} />
      <PointsContainer sliderNodes={sliderNodes} shownType={shownType} />
      <ImageFlexContainer shownType={shownType} isAnimation={animated} />
    </div>
  </>
);

export default SliderPageWrapper;
