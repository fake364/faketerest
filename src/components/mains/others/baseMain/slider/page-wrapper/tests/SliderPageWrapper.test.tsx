import React from 'react';
import { shallow } from 'enzyme';
import SliderPageWrapper from '../SliderPageWrapper';
import { SLIDER_THEMES } from '../../../../../../../common/enums/slider_themes';

describe('SliderPageWrapper', () => {
  it('should render all supplements in slider', () => {
    const wrapper = shallow(
      <SliderPageWrapper
        animated={true}
        shownType={SLIDER_THEMES.HOME_DECOR}
        pageSlidingStopped={false}
        onClickDown={jest.fn()}
        onClickUp={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render modified component with stop class', () => {
    const wrapper = shallow(
      <SliderPageWrapper
        animated={true}
        shownType={SLIDER_THEMES.HOME_DECOR}
        pageSlidingStopped={true}
        onClickDown={jest.fn()}
        onClickUp={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
