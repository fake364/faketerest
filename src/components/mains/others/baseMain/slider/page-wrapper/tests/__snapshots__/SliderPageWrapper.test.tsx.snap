// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderPageWrapper should render all supplements in slider 1`] = `
<Fragment>
  <BouncingButtons
    isAnimated={true}
    onClickDown={[MockFunction]}
    onClickUp={[MockFunction]}
    shownType="homeDecor"
  />
  <div
    className="w-full flex flex-col items-stretch relative z-[-1] animate-slide_bottom"
  >
    <TitleSwitcher
      isAnimation={true}
      shownType="homeDecor"
    />
    <PointsContainer
      shownType="homeDecor"
      sliderNodes={
        Array [
          SliderNode {
            "_type": "dinnerIdea",
          },
          SliderNode {
            "_type": "homeDecor",
          },
          SliderNode {
            "_type": "newOutfit",
          },
          SliderNode {
            "_type": "thumbIdea",
          },
        ]
      }
    />
    <ImageFlexContainer
      isAnimation={true}
      shownType="homeDecor"
    />
  </div>
</Fragment>
`;

exports[`SliderPageWrapper should render modified component with stop class 1`] = `
<Fragment>
  <BouncingButtons
    isAnimated={true}
    onClickDown={[MockFunction]}
    onClickUp={[MockFunction]}
    shownType="homeDecor"
  />
  <div
    className="w-full flex flex-col items-stretch relative z-[-1] animate-slide_bottom stop-animation"
  >
    <TitleSwitcher
      isAnimation={true}
      shownType="homeDecor"
    />
    <PointsContainer
      shownType="homeDecor"
      sliderNodes={
        Array [
          SliderNode {
            "_type": "dinnerIdea",
          },
          SliderNode {
            "_type": "homeDecor",
          },
          SliderNode {
            "_type": "newOutfit",
          },
          SliderNode {
            "_type": "thumbIdea",
          },
        ]
      }
    />
    <ImageFlexContainer
      isAnimation={true}
      shownType="homeDecor"
    />
  </div>
</Fragment>
`;
